alloc/src/utils.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+/// https://doc.rust-lang.org/beta/std/primitive.pointer.html#method.is_aligned_to
```

Comment:
the std lib impl panics when align is not a power of two. And we don't want that right?

Reply:
I think we do. It derives from alignments being a power of two in `Layout`, and since you need a `Layout` to make an allocation in the first place, how did we get to a not-power-of-2?